examples/pipelines/guestbook-resources.yaml

```diff
@@ -1,30 +1,51 @@
 apiVersion: pipeline.knative.dev/v1alpha1
 kind: Resource
 metadata:
-  name: guestbook-resources-sample
+  name: guestbook-resources-git
   namespace: default
 spec:
-  resources:
-    - name: guestbook
-      type: git
-      params:
+    type: git
+    params:
+    - name: url
+      value: github.com/kubernetes/examples
+    - name: revision
+      value: HEAD      
+    - name: serviceAccount
+      value: githubServiceAccount
+---      
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: Resource
+metadata:
+  name: guestbook-resources-redis-docker
+  namespace: default
+spec:
+    type: git
+    params:
       - name: url
-        value: github.com/kubernetes/examples
-      - name: revision
-        value: HEAD      
+        value: github.com/GoogleCloudPlatform/redis-docker/blob/master/4/debian9/4.0/Dockerfile
```

Comment:
hmm, do you think folks would often have `git` resources that point at just one file, or would they be more likely to specify the whole repo?

I'm guessing that even if the goal here was to use this `Dockerfile`, the `Task` would need other files in the repo as well, so maybe this would be something like `value: github.com/GoogleCloudPlatform/redis-docker` and the Task` would also need a param with the path to the Dockerfile, `/4/debian9/4.0/Dockerfile`?

Reply:
yes, you are correct, the url should not include the path to the file